test/e2e/serial/tests/configuration.go

```diff
@@ -1014,13 +1014,21 @@ func accumulateKubeletConfigNames(cms []corev1.ConfigMap) sets.Set[string] {
 
 func waitForMcpsCondition(cli client.Client, ctx context.Context, mcps []*machineconfigv1.MachineConfigPool, condition machineconfigv1.MachineConfigPoolConditionType) error {
 	var eg errgroup.Group
+	interval := configuration.MachineConfigPoolUpdateInterval
+	if condition == machineconfigv1.MachineConfigPoolUpdating {
+		// the transition from updated to updating to updated can be very fast sometimes. so if
+		// the status changed to updating and then to updated while on wait it will miss the updating
+		// status, and it will keep waiting and lastly fail the test. to avoid that decrease the interval
+		// to allow more often checks for the status
+		interval = 2 * time.Second
+	}
```

Comment:
this suggests we may be waiting for the wrong condition. Rather than matching the exact (fragile) sequence of events we should probably do stricter check when we detect `Updated` to make sure we updated to the desired state.

Reply:
correct, I need to look into a way to do that in addition. thanks for raising this

Reply:
I think in such extreme cases we can also watch for the `node`'s annotations:
```
machineconfiguration.openshift.io/currentConfig: rendered-worker-5eb756bcf4dcfb1c4b160b4b1270ab4d
machineconfiguration.openshift.io/desiredConfig: rendered-worker-5eb756bcf4dcfb1c4b160b4b1270ab4d
```
We can have the following steps:
1. Check node desired
2. `Update(MCP)`
3. Wait for node desired and current to change
4. Wait for MCP to move to `Updated` state
This grantees that nodes were actually updated into new revision and MCP is done.

This should be done on all nodes associated with the MCP in parallel in order to save time.

Reply:
yep, very good suggestion. This is a much better form of what I was thinking about, kudos!

Reply:
thanks for your comments, sounds reasonable, I'll verify this scenario

Reply:
implemented in last commit & ready for review